go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go

```diff
@@ -0,0 +1,496 @@
+	"github.com/dolthub/dolt/go/libraries/doltcore/row"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table/editor"
+	"github.com/dolthub/dolt/go/libraries/utils/filesys"
+	"github.com/dolthub/dolt/go/store/hash"
+	"github.com/dolthub/dolt/go/store/prolly"
+	"github.com/dolthub/dolt/go/store/prolly/tree"
+	"github.com/dolthub/dolt/go/store/types"
+	"github.com/dolthub/dolt/go/store/val"
+)
+
+var ErrConfSchIncompatible = errors.New("the conflict schema's columns are not equal to the current schema's columns, please resolve manually")
+
+// DoltConflictsCatFunc runs a `dolt commit` in the SQL context, committing staged changes to head.
+// Deprecated: please use the version in the dprocedures package
+type DoltConflictsCatFunc struct {
+	children []sql.Expression
+}
+
+// NewDoltConflictsResolveFunc creates a new DoltCommitFunc expression whose children represents the args passed in DOLT_CONFLICTS_RESOLVE.
+// Deprecated: please use the version in the dprocedures package
+func NewDoltConflictsResolveFunc(args ...sql.Expression) (sql.Expression, error) {
+	return &DoltConflictsCatFunc{children: args}, nil
+}
+
+func (d DoltConflictsCatFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	args, err := getDoltArgs(ctx, row, d.Children())
+	if err != nil {
+		return 1, err
+	}
+	return DoDoltConflictsResolve(ctx, args)
+}
+
+func getProllyRowMaps(ctx *sql.Context, vrw types.ValueReadWriter, ns tree.NodeStore, hash hash.Hash, tblName string) (prolly.Map, error) {
+	rootVal, err := doltdb.LoadRootValueFromRootIshAddr(ctx, vrw, ns, hash)
+	tbl, ok, err := rootVal.GetTable(ctx, tblName)
+	if err != nil {
+		return prolly.Map{}, err
+	}
+	if !ok {
+		return prolly.Map{}, doltdb.ErrTableNotFound
+	}
+
+	idx, err := tbl.GetRowData(ctx)
+	if err != nil {
+		return prolly.Map{}, err
+	}
+
+	return durable.ProllyMapFromIndex(idx), nil
+}
+
+func resolveProllyConflicts(ctx *sql.Context, tbl *doltdb.Table, tblName string, sch schema.Schema) (*doltdb.Table, error) {
+	var err error
+	artifactIdx, err := tbl.GetArtifacts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	artifactMap := durable.ProllyMapFromArtifactIndex(artifactIdx)
+	iter, err := artifactMap.IterAllConflicts(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutable prolly map
+	ourIdx, err := tbl.GetRowData(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ourMap := durable.ProllyMapFromIndex(ourIdx)
+	mutMap := ourMap.Mutate()
+
+	// accumulate index edits
+	once := true
+	var theirMap prolly.Map
+	var indexEdits []tree.Diff
+	for {
+		cnfArt, err := iter.Next(ctx)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		if once {
+			theirMap, err = getProllyRowMaps(ctx, tbl.ValueReadWriter(), tbl.NodeStore(), cnfArt.TheirRootIsh, tblName)
```

Comment:
|TheirRootIsh| may not be the same for each conflict. You may have to fetch the table more than once. I would store the table outside the loop along with the rootish, if the rootish is different than the one we loaded the table from, reload the table.